Upgrade to XSCL 8.1.1 and support Incremental Copy

blueww opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
1. Upgrade Storage (and related projects) to XSCL 8.1.1 and DMlib 0.5.1 
(DMlib is in release processing, so currently use a private signed one, will replace late)

2. Add a new cmdlet "Start-AzureStorageBlobIncrementalCopy" to support incremental Copy feature

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ Y] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [Y ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ Y] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [ Y] Title of the pull request is clear and informative.
- [ Y] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ N] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module). - XSCL upgrade takes breaking, and May release allows breaking

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [ N/A] Pull request includes test coverage for the included changes.
- [ Y] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [Y ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ Y] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ Y] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ Y] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ Y] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

blueww commented:
On-Demand Test pass:
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1476/

cormacpayne commented on src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobIncrementalCopy.cs:
```diff
@@ -0,0 +1,455 @@
+    using System.Threading.Tasks;
+
+    [Cmdlet(VerbsLifecycle.Start, StorageNouns.IncrementalCopyBlob, SupportsShouldProcess = true, DefaultParameterSetName = ContainerParameterSet),
+       OutputType(typeof(AzureStorageBlob))]
```
@blueww I may have missed it in this cmdlet, but it doesn't look like the cmdlet writes any `AzureStorageBlob` object to the pipeline, even though the output type specifies that it does. It this intended?

blueww replied:
The output in write in this.WriteCloudBlobObject() in below function.

private async Task StartCopyFromUri(long taskId, IStorageBlobManagement destChannel, Uri srcUri, CloudPageBlob destBlob)
        {
            //Don't need to verify the Dest Exist and warn user for overwrite, since incremental Copy won't overwrite the dest blob, but will create a new snapshot for it.
            string copyId = await destChannel.StartIncrementalCopyAsync(destBlob, new CloudPageBlob(srcUri), null, this.RequestOptions, this.OperationContext, this.CmdletCancellationToken).ConfigureAwait(false);
            this.OutputStream.WriteVerbose(taskId, String.Format(Resources.CopyDestinationBlobPending, destBlob.Name, destBlob.Container.Name, copyId));
            this.WriteCloudBlobObject(taskId, destChannel, destBlob);
        }

cormacpayne commented on src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobIncrementalCopy.cs:
```diff
@@ -0,0 +1,455 @@
+
+        [Parameter(HelpMessage = "Blob name", ParameterSetName = ContainerParameterSet, Mandatory = true)]
+        [Parameter(HelpMessage = "Blob name", ParameterSetName = ContainerNameParameterSet, Mandatory = true)]
+        public string SrcBlob
```
@blueww this applies to all cmdlets that have the `Src` and `Dest` prefix:

Can you change the parameter name to include the full word for `Src` and `Dest`? For example, `SrcFoo` will be renamed to `SourceFoo`, and you can add an alias to `SrcFoo` so users can still use it. Same goes for `DestFoo` and `DestinationFoo`.

blueww replied:
This is aligned with orgianl cmdlet Start-AzureStorageBlobCopy.
As no user report the name of Start-AzureStorageBlobCopy is confuse.

Talked with Mark, will add alias to them.

blueww replied:
added

cormacpayne commented on src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobIncrementalCopy.cs:
```diff
@@ -0,0 +1,455 @@
+        [Alias("SrcUri")]
+        [Parameter(HelpMessage = "Source blob uri", Mandatory = true,
+            ValueFromPipelineByPropertyName = true, ParameterSetName = UriParameterSet)]
+        public string AbsoluteUri { get; set; }
```
@blueww please add the `Source` prefix to this parameter

blueww replied:
Will add an alais.

blueww replied:
added

cormacpayne commented on src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobIncrementalCopy.cs:
```diff
@@ -0,0 +1,455 @@
+        [Parameter(HelpMessage = "Source Azure Storage Context Object", ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = BlobToBlobParameterSet)]
+        [Parameter(HelpMessage = "Source Azure Storage Context Object", ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = ContainerParameterSet)]
+        [Parameter(HelpMessage = "Source Azure Storage Context Object", ParameterSetName = UriParameterSet)]
+        public override AzureStorageContext Context { get; set; }
```
@blueww please add the `Source` prefix to this parameter

blueww replied:
Will add Alais.

blueww replied:
added

blueww commented on src/Storage/Commands.Storage/help/Start-AzureStorageBlobIncrementalCopy.md:
````diff
@@ -0,0 +1,370 @@
+```
+
+## DESCRIPTION
+{{Fill in the Description}}
````
added

blueww commented:
I have fixed all comments, and the test is running:
I have updated wxi as following (I have clean and rebuild before update wxi). 
There are 2 kinds of changes. Let me know if you have any comments on that. The wxi change is in CR: CodeFlow
1)	Add “Microsoft.WindowsAzure.Configuration.dll” to storage project. 
2)	Add  “System.Management.Automation.dll” to 4 projects. 

On demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1483/
Sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/980/
